core/src/deserialize.rs

```diff
@@ -30,6 +35,70 @@ macro_rules! deserialize_number {
     };
 }
 
+#[derive(Debug, PartialEq)]
+pub enum EvalOrDeserError {
+    Nickel(crate::error::Error),
+    Deser(RustDeserializationError),
+}
+
+impl<E: Into<crate::error::Error>> From<E> for EvalOrDeserError {
+    fn from(err: E) -> Self {
+        Self::Nickel(err.into())
+    }
+}
+
+impl From<RustDeserializationError> for EvalOrDeserError {
+    fn from(err: RustDeserializationError) -> Self {
+        Self::Deser(err)
+    }
+}
+
+fn from_input<'a, T, Content, Source>(input: Input<Content, Source>) -> Result<T, EvalOrDeserError>
+where
+    T: serde::Deserialize<'a>,
+    Content: std::io::Read,
+    Source: Into<OsString>,
+{
+    Ok(T::deserialize(
+        Program::<CacheImpl>::new_from_input(
+            input,
+            std::io::stderr(),
+            Sink::<(crate::error::Warning, crate::files::Files)>::default(),
```

Comment:
Since you aren't actually emitting warnings, I'd use a `NullReporter` here instead of a `Sink`